### DISCUSSION TITLE
ManiaMap Help

### DISCUSSION THREAD

**Discussion by rorymgrant** [mpewsey/mpewsey.github.io#7] [Q&A] [answered]
Hello,

Can someone please help explain how to run the pipeline and generate a layout? I've done all the earlier steps but I don't know to "run the Run method from a script" and produce a layout. Do I do that in Unity? I'm very new to this so my apologies if the solution is obvious!

**Comment by mpewsey** [comment, marked as answer]
You will need to create a C# MonoBehaviour script in Unity that references the Generation Pipeline object that you plan to use. An example would be:
https://github.com/mpewsey/ManiaMap.Unity/blob/main/Scripts/Runtime/Examples/SampleGenerator.cs

This script is taken from the project samples. It may be useful to open those up and see how they all interconnect. If you go to `Windows > Package Manager`, then the `ManiaMap.Unity` package, you should see `samples` with an import button that you can press to bring them into your project.

![Screenshot 2024-05-24 010147](https://github.com/mpewsey/mpewsey.github.io/assets/23442063/dcdd9768-cd43-46c6-9594-3e123bf97fd4)

From there, the `Sample Generator` or `Sample Layout Map` scenes may be of use to you depending on what you are hoping to do with the layout.
